File: evaluator.fhir/src/main/java/org/opencds/cqf/cql/evaluator/fhir/util/ValidationProfile.java
package org.opencds.cqf.cql.evaluator.fhir.util;

import java.util.List;

public class ValidationProfile {
    private String name;
    private List<String> ignoreKeys;

    public ValidationProfile() { }

    public ValidationProfile(String name, List<String> ignoreKeys) {
        this.name = name;
        this.ignoreKeys = ignoreKeys;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<String> getIgnoreKeys() {
        return ignoreKeys;
    }

    public void setIgnoreKeys(List<String> ignoreKeys) {
        this.ignoreKeys = ignoreKeys;
    }

    public void addIgnoreKey(String key) {
        this.ignoreKeys.add(key);
    }
}

File: cqf-fhir-utility/src/main/java/org/opencds/cqf/fhir/utility/ValidationProfile.kt
package org.opencds.cqf.fhir.utility

class ValidationProfile {
    var name: String? = null
    private var ignoreKeys: MutableList<String?>? = null

    constructor()

    constructor(name: String?, ignoreKeys: MutableList<String?>) {
        this.name = name
        this.ignoreKeys = ignoreKeys
    }

    fun getIgnoreKeys(): MutableList<String?> {
        return ignoreKeys!!
    }

    fun setIgnoreKeys(ignoreKeys: MutableList<String?>) {
        this.ignoreKeys = ignoreKeys
    }

    fun addIgnoreKey(key: String?) {
        this.ignoreKeys!!.add(key)
    }
}
